src/modules/claim/claim.service.ts

```diff
@@ -75,73 +69,149 @@ export class ClaimService {
+  }
+
+  /**
+   * Handles claim issuance request saving and updates.
+   * Two scenarios are handled - issue requested claim and issue not-requested claim
+   * @param rq IClaimIssuance request
    */
-  public async handleExchangeMessage(
-    data: IClaimIssuance | IClaimRejection | IClaimRequest,
-  ): Promise<string> {
-    try {
-      const claim: RoleClaim = await this.getById(data.id);
-      if (!claim && 'token' in data) {
-        const {
-          claimData: { claimType, claimTypeVersion },
-        } = jwt.decode(data.token) as DecodedClaimToken;
-
-        const dto = await ClaimRequestDTO.create({
-          ...data,
-          claimType,
-          claimTypeVersion,
-        });
-
-        await this.roleService.verifyEnrolmentPrecondition({
-          claimType,
-          userDID: dto.requester,
-        });
-
-        await this.create(dto);
-        return;
-      }
-      if (!claim && 'issuedToken' in data) {
-        const {
-          claimData: { claimType, claimTypeVersion },
-        } = jwt.decode(data.issuedToken) as DecodedClaimToken;
-
-        const dto = await NewClaimIssueDTO.create({
-          ...data,
-          claimType,
-          claimTypeVersion,
-        });
-
-        await this.roleService.verifyEnrolmentPrecondition({
-          claimType,
-          userDID: dto.requester,
-        });
-
-        await this.createAndIssue(dto);
-        return;
-      }
-      if (claim && !claim.isAccepted && 'isRejected' in data) {
-        const dto = await ClaimRejectionDTO.create(data);
-
-        await this.reject(dto.id);
-        return;
-      }
-
-      if (
-        (claim && !claim.isRejected && 'issuedToken' in data) ||
-        'onChainProof' in data
-      ) {
-        const dto = await ClaimIssueDTO.create(data);
-
-        await this.issue(dto);
-        return;
-      }
-    } catch (err) {
-      this.logger.error(err);
+  public async handleClaimIssuanceRequest(rq: IClaimIssuance) {
+    const claim: RoleClaim = await this.getById(rq.id);
+    const {
+      claimData: { claimType, claimTypeVersion },
+    } = jwt.decode(rq.issuedToken) as DecodedClaimToken;
+
+    if (!claim && !!rq.issuedToken) {
+      const dto = await NewClaimIssueDTO.create({
+        ...rq,
+        claimType,
+        claimTypeVersion,
+      });
+
+      await this.roleService.verifyEnrolmentPrecondition({
+        claimType,
+        userDID: dto.requester,
+      });
+
+      await this.createAndIssue(dto);
+    } else if (
+      claim &&
+      !claim.isRejected &&
+      (!!rq.issuedToken || !!rq.onChainProof)
+    ) {
+      const dto = await ClaimIssueDTO.create(rq);
+      await this.issue(dto);
     }
   }
 
+  /**
+   * Handles claim rejection request saving and updates.
+   * @param rq IClaimRejection request
+   */
+  public async handleClaimRejectionRequest(rq: IClaimRejection) {
+    const claim: RoleClaim = await this.getById(rq.id);
+    if (!claim || claim.isAccepted || !rq.isRejected) return;
```

Comment:
What do you think of returning some error result here so that the server can return a `400 bad request`? I think that all of these checks are for an incorrect request from the DID that made the request?